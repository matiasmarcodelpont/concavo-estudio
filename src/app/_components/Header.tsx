'use client'

import Logo from '@/components/domain/Logo'
import Link from 'next/link'
import { NavBarLayout } from './NavBarLayout'
import { Ambiente, OmitReferences } from '@/data/types'
import { usePathname } from 'next/navigation'
import { cn } from '@/lib/utils'

interface HeaderProps {
  ambientes: OmitReferences<Omit<Ambiente, 'images'>>[]
}

export const Header = ({ ambientes }: HeaderProps) => {
  const pathname = usePathname()

  return (
    <header className='fixed top-0 w-full bg-bone flex justify-between px-2 sm:px-6 py-4 items-center z-10'>
      <NavBarLayout ambientes={ambientes} />

      <h1 className={cn(pathname == '/' ? 'invisible' : '', 'my-0')}>
        <Link href='/'>
          <Logo />
        </Link>
      </h1>

      <div className='w-[40px]' />
    </header>
  )
}
