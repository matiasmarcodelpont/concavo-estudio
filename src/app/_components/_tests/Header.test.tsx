import { render, within } from '@testing-library/react'

import { Header } from '../Header'

describe('Header', () => {
  it('matches snapshot', () => {
    const { getByRole } = render(<Header />)
    const component = getByRole('banner')

    expect(component).toMatchSnapshot()
  })

  it('renders the logo', () => {
    const { getByRole } = render(<Header />)

    const logoWrapper = getByRole('link')
    within(logoWrapper).getByRole('img')
  })
})
