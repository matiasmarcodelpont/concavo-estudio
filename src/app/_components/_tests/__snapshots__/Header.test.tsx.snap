// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header
  class="px-4 flex items-center gap-2 sticky top-0 z-40 bg-bone shadow"
>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    aria-label="Open navigation menu"
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-slate-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 dark:ring-offset-slate-950 dark:focus-visible:ring-slate-300 hover:bg-slate-100 hover:text-slate-900 dark:hover:bg-slate-800 dark:hover:text-slate-50 h-10 w-10"
    data-state="closed"
    type="button"
  >
    <svg
      class="lucide lucide-equal size-6 sm:size-9"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="9"
        y2="9"
      />
      <line
        x1="5"
        x2="19"
        y1="15"
        y2="15"
      />
    </svg>
  </button>
  <a
    class="uppercase text-lg text-center flex flex-col items-center m-auto p-4 md:p-6"
    href="/"
  >
    <div
      class="w-[150px] h-[50px] sm:w-[201px] sm:h-[67px] relative"
    >
      <img
        alt="Logo CÃ³ncavo"
        class="object-contain"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Flogos%2Fconcavo_black.png&w=3840&q=75"
        srcset="/_next/image?url=%2Flogos%2Fconcavo_black.png&w=640&q=75 640w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=750&q=75 750w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=828&q=75 828w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=1080&q=75 1080w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=1200&q=75 1200w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=1920&q=75 1920w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=2048&q=75 2048w, /_next/image?url=%2Flogos%2Fconcavo_black.png&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
  </a>
  <div
    class="flex-shrink-0 w-[40px] h-[10px] bg-mustard sm:size-9 invisible"
  />
</header>
`;
